hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -461,39 +466,7 @@ def arrays(
     return ArrayStrategy(elements, shape, dtype, fill, unique)
 
 
-@defines_strategy()
-def array_shapes(
-    *,
-    min_dims: int = 1,
-    max_dims: Optional[int] = None,
-    min_side: int = 1,
-    max_side: Optional[int] = None,
-) -> st.SearchStrategy[Shape]:
-    """Return a strategy for array shapes (tuples of int >= 1)."""
-    check_type(int, min_dims, "min_dims")
-    check_type(int, min_side, "min_side")
-    if min_dims > 32:
-        raise InvalidArgument(
-            "Got min_dims=%r, but numpy does not support arrays greater than 32 dimensions"
-            % min_dims
-        )
-    if max_dims is None:
-        max_dims = min(min_dims + 2, 32)
-    check_type(int, max_dims, "max_dims")
-    if max_dims > 32:
-        raise InvalidArgument(
-            "Got max_dims=%r, but numpy does not support arrays greater than 32 dimensions"
-            % max_dims
-        )
-    if max_side is None:
-        max_side = min_side + 5
-    check_type(int, max_side, "max_side")
-    order_check("dims", 0, min_dims, max_dims)
-    order_check("side", 0, min_side, max_side)
-
-    return st.lists(
-        st.integers(min_side, max_side), min_size=min_dims, max_size=max_dims
-    ).map(tuple)
+array_shapes = _array_helpers.array_shapes
```

Comment:
Just `from _array_helpers import array_shapes` for this pattern; it's exactly equivalent with less code.